### PR TITLE
Generate Java source file for Android from the list of logs.

### PR DESCRIPTION
**Pull request by plietar**
This file will be included in conscrypt for Android

### CODE REVIEW

**Comment by plietar** [comment]
Generated output :
https://android-review.googlesource.com/#/c/171932/6/src/main/java/org/conscrypt/ct/KnownLogs.java

DO NOT MERGE YET
I may need to change the format based on reviews on the Android side.

File: python/utilities/log_list/print_log_list.py [hunk 1]
```diff
@@ -131,6 +134,75 @@ def generate_code_for_chromium(json_log_list, output_file):
     f.write("\nconst size_t kNumKnownCTLogs = %d;\n" % len(logs))
     write_cpp_footer(f, include_guard)
 
+def write_android_header(f):
```
Comment (eranmes) [inline python/utilities/log_list/print_log_list.py]: Break this into a separate module (within this directory).

Comment (eranmes) [inline python/utilities/log_list/print_log_list.py]: While you're at it, please look into breaking the C++ code generation to a module of its own.

**Comment by plietar** [comment]
I've changed the output format to be more generic rather than just file conscrypt.
The package and class names still default to conscrypt, but can be overridden with a flag.
The copyright notice is still the one from AOSP.

Also split out the code generation is separate modules

File: python/utilities/log_list/java_generator.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+def _write_java_class(f, logs, cls):
+    descriptions = ('        "%s",\n' % log["description"] for log in logs)
+    urls = ('        "%s",\n' % log["url"] for log in logs)
+    keys = (_encode_key(log["description"], base64.decodestring(log["key"]))
+            for log in logs)
+
+    f.write(
+        "public final class %(cls)s {\n"
+        "    public static final int LOG_COUNT = %(count)d;\n"
+        "    public static final String[] LOG_DESCRIPTIONS = new String[] {\n"
```
Comment (eranmes) [inline python/utilities/log_list/java_generator.py]: Seems like some lines here are too long, please run it through pylint.

File: python/utilities/log_list/java_generator.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        "package %(package)s;\n\n" %
+        {"year": year,
+         "package": package})
+
+def _encode_key(description, key):
+    unsigned = (ord(c) for c in key)
+    signed = (u - 256 if u > 127 else u for u in unsigned)
+    return ("        // " + description + "\n"
+            "        new byte[] {\n"
+            + textwrap.fill(", ".join("%d" % s for s in signed), width=85,
```
Comment (eranmes) [inline python/utilities/log_list/java_generator.py]: Could you assign the result from this call to a variable you'll include in the string generated here, so the code is slightly more readable?

File: python/utilities/log_list/java_generator.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+        "%(descriptions)s"
+        "    };\n"
+        "    public static final String[] LOG_URLS = new String[] {\n"
+        "%(urls)s"
+        "    };\n"
+        "    public static final byte[][] LOG_KEYS = new byte[][] {\n"
+        "%(keys)s"
+        "    };\n"
+        "}\n" % {
+            "cls": cls,
```
Comment (eranmes) [inline python/utilities/log_list/java_generator.py]: nit: Prefer a long name than a shorthand.

**Comment by eranmes** [comment]
A few minor comments, LGTM otherwise - please rebase and let me know so I'll merge and commit.

**Pull request merged**